pkg/recipes/configloader/secrets.go

```diff
@@ -34,21 +35,55 @@ type secretsLoader struct {
 	ArmClientOptions *arm.ClientOptions
 }
 
-func (e *secretsLoader) LoadSecrets(ctx context.Context, secretStore string) (v20231001preview.SecretStoresClientListSecretsResponse, error) {
-	secretStoreID, err := resources.ParseResource(secretStore)
-	if err != nil {
-		return v20231001preview.SecretStoresClientListSecretsResponse{}, err
+// LoadSecrets loads secrets from secret stores based on input map of provided secret store IDs and secret keys.
+// It returns a map of secret data, where the keys are the secret store IDs and the values are maps of secret keys and their corresponding values.
+func (e *secretsLoader) LoadSecrets(ctx context.Context, secretStoreIDResourceKeys map[string][]string) (secretData map[string]map[string]string, err error) {
+	for secretStoreID, secretKeys := range secretStoreIDResourceKeys {
+		secretStoreResourceID, err := resources.ParseResource(secretStoreID)
+		if err != nil {
+			return nil, err
+		}
+
+		client, err := v20231001preview.NewSecretStoresClient(secretStoreResourceID.RootScope(), &aztoken.AnonymousCredential{}, e.ArmClientOptions)
+		if err != nil {
+			return nil, err
+		}
+
+		// Retrieve the secrets from the secret store.
+		secrets, err := client.ListSecrets(ctx, secretStoreResourceID.Name(), map[string]any{}, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		// Populate the secretData map.
+		secretData, err = populateSecretData(secretStoreID, secretKeys, &secrets)
+		if err != nil {
+			return nil, err
+		}
 	}
 
-	client, err := v20231001preview.NewSecretStoresClient(secretStoreID.RootScope(), &aztoken.AnonymousCredential{}, e.ArmClientOptions)
-	if err != nil {
-		return v20231001preview.SecretStoresClientListSecretsResponse{}, err
+	return secretData, nil
+}
+
+// populateSecretData is a helper function to populate secret data from a secret store.
+func populateSecretData(secretStoreID string, secretKeys []string, secrets *v20231001preview.SecretStoresClientListSecretsResponse) (map[string]map[string]string, error) {
+	secretData := make(map[string]map[string]string)
+
+	if secrets == nil {
+		return nil, fmt.Errorf("secrets not found for secret store ID '%s'", secretStoreID)
 	}
 
-	secrets, err := client.ListSecrets(ctx, secretStoreID.Name(), map[string]any{}, nil)
-	if err != nil {
-		return v20231001preview.SecretStoresClientListSecretsResponse{}, err
+	for _, secretKey := range secretKeys {
+		secretDataValue, ok := secrets.Data[secretKey]
+		if ok {
```

Comment:
Why not inline this so that it can look clearer?

Reply:
updated